core/process.py

```diff
@@ -868,22 +889,19 @@ def MaybeClosePipe(self):
       posix.close(self.close_r)
       posix.close(self.close_w)
 
-  def Start(self, why):
-    # type: (trace_t) -> int
-    """Start this process with fork(), handling redirects."""
-    # TODO: If OSH were a job control shell, we might need to call some of
-    # these here.  They control the distribution of signals, some of which
-    # originate from a terminal.  All the processes in a pipeline should be in
-    # a single process group.
-    #
-    # - posix.setpgid()
-    # - posix.setpgrp() 
-    # - posix.tcsetpgrp()
-    #
-    # NOTE: posix.setsid() isn't called by the shell; it's should be called by the
-    # login program that starts the shell.
-    #
-    # The whole job control mechanism is complicated and hacky.
+  def Start(self, why, pgrp=-1):
+    # type: (trace_t, int) -> int
+    """
+    Start this process with fork(), handling redirects.
+    If job control is enabled and pgrp is positive, the process will be placed in the provided group.
+    """
+    if self.job_state.fg_pipeline:
+      self.parent_pipeline = self.job_state.fg_pipeline
+      pgrp = self.parent_pipeline.group_id
+
+    # Command substitution should run in its parent shell's group.
+    if why.tag_() == trace_e.CommandSub:
+      pgrp = posix.getpgid(0)
```

Comment:
I can't shake the feeling that this special case should just "fall out" of the default behavior somehow... Gonna keep spinning my wheels on this